runtime-testsuite/test/org/antlr/v4/test/runtime/java/api/perf/TimeLexerSpeed.java

```diff
@@ -0,0 +1,162 @@
+		tests.legacy_grapheme_utf8("udhr_hin.txt", n, false);
+		tests.legacy_grapheme_utf8("udhr_hin.txt", n, true);
+		// legacy can't handle the emoji (32 bit stuff)
+
+		tests.new_grapheme_utf8("udhr_kor.txt", n, false);
+		tests.new_grapheme_utf8("udhr_kor.txt", n, true);
+		tests.new_grapheme_utf8("udhr_hin.txt", n, false);
+		tests.new_grapheme_utf8("udhr_hin.txt", n, true);
+		tests.new_grapheme_utf8("emoji.txt", n, false);
+		tests.new_grapheme_utf8("emoji.txt", n, true);
+	}
+
+	public void legacy_java_ascii(int n, boolean clearLexerDFACache) throws Exception {
+		URL sampleJavaFile = TimeLexerSpeed.class.getClassLoader().getResource(Parser_java_file);
+		CharStream input = new ANTLRFileStream(sampleJavaFile.getFile());
+		JavaLexer lexer = new JavaLexer(input);
+		double avg = tokenize(lexer, n, clearLexerDFACache);
+		String currentMethodName = new Exception().getStackTrace()[0].getMethodName();
+		System.out.printf("%25s average time %5dus over %4d runs of %5d symbols%s\n",
+		                  currentMethodName,
+		                  (int)avg,
+		                  n,
+		                  input.size(),
+		                  clearLexerDFACache ? " DFA cleared" : "");
+	}
+
+	public void new_java_utf8(int n, boolean clearLexerDFACache) throws Exception {
+		URL sampleJavaFile = TimeLexerSpeed.class.getClassLoader().getResource(Parser_java_file);
+		CharStream input = CharStreams.fromPath(Paths.get(sampleJavaFile.getFile()));
+		JavaLexer lexer = new JavaLexer(input);
+		double avg = tokenize(lexer, n, clearLexerDFACache);
+		String currentMethodName = new Exception().getStackTrace()[0].getMethodName();
+		System.out.printf("%25s average time %5dus over %4d runs of %5d symbols%s\n",
+		                  currentMethodName,
+		                  (int)avg,
+		                  n,
+		                  input.size(),
+		                  clearLexerDFACache ? " DFA cleared" : "");
+	}
+
+	public void legacy_grapheme_utf8(String fileName, int n, boolean clearLexerDFACache) throws Exception {
+		URL sampleJavaFile = TimeLexerSpeed.class.getClassLoader().getResource(PerfDir+"/"+fileName);
+		CharStream input = new ANTLRFileStream(sampleJavaFile.getFile());
+		graphemesLexer lexer = new graphemesLexer(input);
+		double avg = tokenize(lexer, n, clearLexerDFACache);
+		String currentMethodName = new Exception().getStackTrace()[0].getMethodName();
+		System.out.printf("%25s average time %5dus over %4d runs of %5d symbols from %s%s\n",
+		                  currentMethodName,
+		                  (int)avg,
+		                  n,
+		                  input.size(),
+		                  fileName,
+		                  clearLexerDFACache ? " DFA cleared" : "");
+	}
+
+	public void new_grapheme_utf8(String fileName, int n, boolean clearLexerDFACache) throws Exception {
+		URL sampleJavaFile = TimeLexerSpeed.class.getClassLoader().getResource(PerfDir+"/"+fileName);
+		CharStream input = CharStreams.fromPath(Paths.get(sampleJavaFile.getFile()));
```

Comment:
Actually, i meant these to test lexing time so it's ok. I'll separately test the stream load time.